src/Client/Adapter/Curl.php

```diff
@@ -209,6 +209,13 @@ public function connect($host, $port = 80, $secure = false)
             }
         }
 
+        if (isset($this->config['sslcafile']) && $this->config['sslcafile']) {
```

Comment:
Any reason to do that? I would do the opposite: avoid setting the `CURLOPT_CAINFO` and `CURLOPT_CAPATH` curl options if `CURLOPT_SSL_VERIFYPEER` is false (that because I don't know how the underlying libcurl behaves in that case...)